pkg/stores/partition/listprocessor/processor.go

```diff
@@ -0,0 +1,300 @@
+	primaryOrder   SortOrder
+	secondaryOrder SortOrder
+}
+
+// String returns the sort parameters as a query string.
+func (s Sort) String() string {
+	field := ""
+	if s.primaryOrder == DESC {
+		field = "-" + field
+	}
+	field += strings.Join(s.primaryField, ".")
+	if len(s.secondaryField) > 0 {
+		field += ","
+		if s.secondaryOrder == DESC {
+			field += "-"
+		}
+		field += strings.Join(s.secondaryField, ".")
+	}
+	return field
+}
+
+// Pagination represents how to return paginated results.
+type Pagination struct {
+	pageSize int
+	page     int
+}
+
+// PageSize returns the integer page size.
+func (p Pagination) PageSize() int {
+	return p.pageSize
+}
+
+// ParseQuery parses the query params of a request and returns a ListOptions.
+func ParseQuery(apiOp *types.APIRequest) *ListOptions {
+	chunkSize := getLimit(apiOp)
+	q := apiOp.Request.URL.Query()
+	cont := q.Get(continueParam)
+	filterParams := q[filterParam]
+	filterOpts := []Filter{}
+	for _, filters := range filterParams {
+		filter := strings.Split(filters, "=")
+		if len(filter) != 2 {
+			continue
+		}
+		filterOpts = append(filterOpts, Filter{field: strings.Split(filter[0], "."), match: filter[1]})
+	}
+	// sort the filter fields so they can be used as a cache key in the store
+	sort.Slice(filterOpts, func(i, j int) bool {
+		fieldI := strings.Join(filterOpts[i].field, ".")
+		fieldJ := strings.Join(filterOpts[j].field, ".")
+		return fieldI < fieldJ
+	})
+	sortOpts := Sort{}
+	sortKeys := q.Get(sortParam)
+	if sortKeys != "" {
+		sortParts := strings.SplitN(sortKeys, ",", 2)
+		primaryField := sortParts[0]
+		if primaryField != "" && primaryField[0] == '-' {
+			sortOpts.primaryOrder = DESC
+			primaryField = primaryField[1:]
+		}
+		if primaryField != "" {
+			sortOpts.primaryField = strings.Split(primaryField, ".")
+		}
+		if len(sortParts) > 1 {
+			secondaryField := sortParts[1]
+			if secondaryField != "" && secondaryField[0] == '-' {
+				sortOpts.secondaryOrder = DESC
+				secondaryField = secondaryField[1:]
+			}
+			if secondaryField != "" {
+				sortOpts.secondaryField = strings.Split(secondaryField, ".")
+			}
+		}
+	}
+	var err error
+	pagination := Pagination{}
+	pagination.pageSize, err = strconv.Atoi(q.Get(pageSizeParam))
+	if err != nil {
+		pagination.pageSize = 0
+	}
+	pagination.page, err = strconv.Atoi(q.Get(pageParam))
+	if err != nil {
+		pagination.page = 1
+	}
+	revision := q.Get(revisionParam)
+	return &ListOptions{
+		ChunkSize:  chunkSize,
+		Resume:     cont,
+		Filters:    filterOpts,
+		Sort:       sortOpts,
+		Pagination: pagination,
+		Revision:   revision,
+	}
+}
+
+// getLimit extracts the limit parameter from the request or sets a default of 100000.
+// The default limit can be explicitly disabled by setting it to a negative value.
+// If the default is accepted, clients must be aware that the list may be incomplete, and use the "continue" token to get the next chunk of results.
+func getLimit(apiOp *types.APIRequest) int {
```

Comment:
We should either update our behavior to state that if the limit is 0 or negative we unset the limit parameter or leave the check that overrides `0`to be the `defaultLimit`. I couldn't find any official documentation on `limit=0` in the kubeapi references, but through some simple curl testing, if I add  `limit=0` I get back all the results. This could be unexpected for users since our previous implementation would override the limit if it was set to '0'